.github/workflows/test-python3-support.yml

```diff
@@ -0,0 +1,133 @@
+            gradle \
+            libcap-dev \
+            libcurl4-openssl-dev \
+            libdbd-sqlite3 \
+            libdbi0-dev \
+            libesmtp-dev \
+            libgeoip-dev \
+            libglib2.0-dev \
+            libhiredis-dev \
+            libivykis-dev \
+            librabbitmq-dev \
+            libmongoc-dev \
+            libnet1-dev \
+            libriemann-client-dev \
+            librdkafka-dev \
+            libwrap0-dev \
+            pkg-config \
+            sqlite3 \
+            xsltproc \
+            criterion-dev \
+            libmaxminddb-dev \
+            libxml2-utils \
+            doxygen \
+            libsnmp-dev \
+            python3-pip \
+            python3-setuptools
+
+          python3 -m pip install --user -r requirements.txt
+
+      - name: autogen
+        run:
+          set -e
+          export PYTHONUSERBASE=$HOME/python_packages
+          export PKG_CONFIG_PATH=/usr/local/opt/openssl/lib/pkgconfig:$PKG_CONFIG_PATH
+          export PATH=/usr/local/opt/bison/bin:/usr/local/opt/libnet/bin:$PYTHONUSERBASE/bin:$PATH
+
+          ./autogen.sh
+
+      - name: configure
+        run: |
+          set -e
+          export PYTHONUSERBASE=$HOME/python_packages
+          export PKG_CONFIG_PATH=/usr/local/opt/openssl/lib/pkgconfig:$PKG_CONFIG_PATH
+          export PATH=/usr/local/opt/bison/bin:/usr/local/opt/libnet/bin:$PYTHONUSERBASE/bin:$PATH
+
+          ./configure \
+            --with-python=3 \
+            --with-ivykis=system \
+            --disable-sun-streams \
+            --disable-systemd \
+            --disable-pacct \
+            --disable-smtp \
+            --disable-geoip \
+            --enable-all-modules \
+            |& tee -i configure.log
+
+      - name: "artifact: configure.log"
+        uses: actions/upload-artifact@v1
+        if: failure()
+        with:
+          name: configure.log
+          path: configure.log
+
+      - name: build
+        run: |
+          set -e
+          export PYTHONUSERBASE=$HOME/python_packages
+          export PKG_CONFIG_PATH=/usr/local/opt/openssl/lib/pkgconfig:$PKG_CONFIG_PATH
+          export PATH=/usr/local/opt/bison/bin:/usr/local/opt/libnet/bin:$PYTHONUSERBASE/bin:$PATH
+
+          make --keep-going -j $(sysctl -n hw.physicalcpu) || \
+            { \
+              S=$?; \
+              make V=1 |& tee -i build.log; \
+              exit $S; \
+            }
+
+      - name: "artifact: build.log"
+        uses: actions/upload-artifact@v1
+        if: failure()
+        with:
+          name: build.log
+          path: build.log
+
+      - name: unit-test
+        run: |
+          set -e
+          export PYTHONUSERBASE=$HOME/python_packages
+          export PKG_CONFIG_PATH=/usr/local/opt/openssl/lib/pkgconfig:$PKG_CONFIG_PATH
+          export PATH=/usr/local/opt/bison/bin:/usr/local/opt/libnet/bin:$PYTHONUSERBASE/bin:$PATH
+
+          make --keep-going check -j $(sysctl -n hw.physicalcpu) || \
```

Comment:
There is an error in the log: `sysctl: cannot stat /proc/sys/hw/physicalcpu: No such file or directory`

Reply:
Wow nice catch, thanks!
I will just remove it, I guess. What do you think?

Reply:
It would be nice to add a reasonable limit. I would probably quickly fix with `-j 4` now, without knowing much about the build environment.

Reply:
https://help.github.com/en/actions/automating-your-workflow-with-github-actions/virtual-environments-for-github-hosted-runners#about-github-hosted-runners
The link says they offer Standard_DS2_v2.

Specs:
https://docs.microsoft.com/en-us/azure/virtual-machines/windows/sizes-general#dsv2-series

Standard_DS2_v2 shows 2 vCPU, so `-j2` ?

Reply:
I think I will go with `-j 3` for ... reasons.

Reply:
[optional] suggest `nproc` (that sysctl stuff is for macos)

Reply:
*facepalm*